tests/e2e/utils/kubeflow_installation.py

```diff
@@ -0,0 +1,200 @@
+        identifier = path_dic[component_name]["validations"]["identifier"]
+        common_label = path_dic[component_name]["validations"]["common_label"]
+        namespace = path_dic[component_name]["validations"]["namespace"]
+        print(f"Waiting for {component_name} pods to be ready ...")
+        retcode = kubectl_wait_pods(common_label, namespace, identifier)
+        assert retcode == 0
+    print(f"All {component_name} pods are running!")
+
+
+def build_certmanager():
+    retcode = subprocess.call(
+        f"helm repo add jetstack https://charts.jetstack.io".split()
+    )
+    assert retcode == 0
+    retcode = subprocess.call(f"helm repo update".split())
+    assert retcode == 0
+    cmd = f"helm install cert-manager jetstack/cert-manager \
+                        --namespace cert-manager \
+                        --create-namespace \
+                        --version v1.9.1 \
+                        --set installCRDs=true".split()
+    return subprocess.call(cmd)
+
+
+def build_alb_controller(cluster_name):
+    retcode = subprocess.call(
+        f"helm repo add eks https://aws.github.io/eks-charts".split()
+    )
+    assert retcode == 0
+    retcode = subprocess.call(f"helm repo update".split())
+    assert retcode == 0
+    cmd = f"helm install aws-load-balancer-controller eks/aws-load-balancer-controller \
+            -n kube-system \
+            --set clusterName={cluster_name} \
+            --set serviceAccount.create=false \
+            --set serviceAccount.name=aws-load-balancer-controller \
+            --version v2.4.3".split()
+    return subprocess.call(cmd)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    INSTALLATION_OPTION_DEFAULT = "kustomize"
+    parser.add_argument(
+        "--installation_option",
+        type=str,
+        default=INSTALLATION_OPTION_DEFAULT,
+        help=f"Kubeflow Installation option (helm/kustomize), default is set to {INSTALLATION_OPTION_DEFAULT}",
+        required=False,
+    )
+    AWS_TELEMETRY_DEFAULT = "enable"
+    parser.add_argument(
+        "--aws_telemetry_option",
+        type=str,
+        default=AWS_TELEMETRY_DEFAULT,
+        help=f"Usage tracking (enable/disable), default is set to {AWS_TELEMETRY_DEFAULT}",
+        required=False,
+    )
+    DEPLOYMENT_OPTION_DEFAULT = "vanilla"
+    parser.add_argument(
+        "--deployment_option",
+        type=str,
+        default=DEPLOYMENT_OPTION_DEFAULT,
+        help=f"Kubeflow deployment options (vanilla/cognito/rds_and_s3/rds_only/s3_only/cognito-rds-s3), default is set to {DEPLOYMENT_OPTION_DEFAULT}",
+        required=False,
+    )
+
+    args, _ = parser.parse_known_args()
+    INSTALLATION_OPTION = args.installation_option
+    AWS_TELEMETRY_OPTION = args.aws_telemetry_option
+    DEPLOYMENT_OPTION = args.deployment_option
+    CLUSTER_NAME = os.environ["CLUSTER_NAME"]
+    install_kubeflow(
+        INSTALLATION_OPTION, AWS_TELEMETRY_OPTION, DEPLOYMENT_OPTION, CLUSTER_NAME
```

Comment:
It is used in the integration test. Since we are calling the installation script in the fixture on_creation, thus they are not as parsers in that case.

in the new installation.py fixture (tests/e2e/fixtures/installation.py) to replace kustomize.py

the cluster input is used for cognito integration test for helm load balancer component
```
@pytest.fixture(scope="class")
def installation(
    metadata, aws_telemetry_option, deployment_option, cluster, clone_upstream, configure_manifests, installation_path, installation_option, request
):
    """
    This fixture is created once for each test class.

    Before all tests are run, installs kubeflow using the manifest at `kustomize_path`
    if `kustomize_path` was not provided in the metadata.

    After all tests are run, uninstalls kubeflow using the manifest at `kustomize_path`
    if the flag `--keepsuccess` was not provided as a pytest argument.
    """

    def on_create():
        install_kubeflow(installation_option, aws_telemetry_option, deployment_option, cluster)
        
```

Reply:
I mean you directly do `install_kubeflow(args., args.)`?

Reply:
addressed.